src/main/webapp/app/shared/util/pagination-crud-store.ts

```diff
@@ -4,14 +4,19 @@ import BaseCrudStore from 'app/shared/util/base-crud-store';
 import _ from 'lodash';
 
 export const debouncedSearchWithPagination = _.debounce(
-  (query, page, size, sort, searchEntities) =>
-    searchEntities({
-      query,
-      page,
-      size,
-      sort,
-    }),
-  500,
+  (query: string, page, size, sort, searchEntities) => {
+    if (query && query.length > 2) {
```

Comment:
We could show a message to user that search has to be at least 3 characters. I can look into this in my PR to replace OncoKbTable in transcript with public.